src/main/java/com/google/devtools/build/lib/bazel/repository/GitCloner.java

```diff
@@ -181,6 +189,15 @@ public static SkyValue clone(Rule rule, Path outputDirectory, EventHandler event
     return new HttpDownloadValue(descriptor.directory);
   }
 
+  private static void setProxyIfNeeded() throws IOException {
+	if(!"".equalsIgnoreCase(System.getenv("HTTP_PROXY"))){
```

Comment:
For now you could create an instance and pass it in via the relevant constructors in https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/BazelRepositoryModule.java#L82-94.